ccp-openshift-scan/scanning/misc_package_updates.py

```diff
@@ -0,0 +1,140 @@
+#!/usr/bin/env python
+
+# this scan file has utilities to find pip, npm, gem package updates
+
+import lib
+
+import sys
+
+
+def binary_does_not_exist(response):
+    """
+    Used to figure if the npm, pip, gem binary exists in the container image
+    """
+    if 'executable file not found in' in response or \
+            'not found' in response or \
+            'No such file or directory' in response:
+        return True
+    return False
+
+
+def find_pip_updates(executable="/usr/bin/pip"):
+    """
+    Finds out outdated installed packages of pip
+    """
+    command = [executable, "list", "--outdated", "--disable-pip-version-check"]
+    out, err = [], ""
+
+    try:
+        out, err = lib.run_cmd_out_err(command)
+    except Exception as e:
+        err = e
+
+    if err:
+        if binary_does_not_exist(err):
+            return "{} is not installed".format(executable)
+        else:
+            return "Failed to find the pip updates."
+    else:
+        if out.strip():
+            return out.strip().split("\n")
+        else:
+            return []
+
+
+def find_npm_updates(executable="/usr/bin/npm"):
+    """
+    Finds out outdated installed packages of npm
+    """
+    command = [executable, "-g", "outdated"]
+    out, err = [], ""
+
+    try:
+        out, err = lib.run_cmd_out_err(command)
+    except Exception as e:
+        err = e
+
+    if err:
+        if binary_does_not_exist(err):
+            return "{} is not installed".format(executable)
+        else:
+            return "Failed to find the npm updates."
+    else:
+        if out.strip():
+            return out.strip().split("\n")
+        else:
+            return []
+
+
+def find_gem_updates(executable="/usr/bin/gem"):
+    """
+    Finds out outdated installed packages of gem
+    """
+    command = [executable, "outdated"]
+    out, err = [], ""
+
+    try:
+        out, err = lib.run_cmd_out_err(command)
+    except Exception as e:
+        err = e
+
+    if err:
+        if binary_does_not_exist(err):
+            return "{} is not installed".format(executable)
+        else:
+            return "Failed to find the gem updates."
+    else:
+        if out.strip():
+            return out.strip().split("\n")
+        else:
+            return []
+
+
+def print_updates(binary):
```

Comment:
I used the name as binary because of its value like `/usr/bin/pip` or `/usr/bin/npm`. Throughout the scanner, we have used the term binary, like `binary_does_not_exist` in function names, etc so just keeping the consistency.

Reply:
When do `print_updates(binary)` it sounds like we want to print updates for the binary. But when we do `print_updates(manager)` it sounds like we want to print updates for the package manager. Neither is perfect but latter sounds closer to what we are trying to do.

Reply:
@navidshaikh Can you reply?

Reply:
IMO - binary is the command line of the manager, that's why I kept it so.

Reply:
Agreed that binary is the command line of the manager. But is it the binary whose updates we're trying to find? No. We're finding the updates for packages installed using particular package manager.

Reply:
@navidshaikh can you answer this?